<launch>
  <!-- Arguments -->
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="cmd_vel_topic" default="/cmd_vel" />
  <arg name="odom_topic" default="odom" />
  <arg name="move_forward_only" default="false"/>

  <!-- move_base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
    <rosparam file="$(find minitask5)/param/costmap_common_params_$(arg model).yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find minitask5)/param/costmap_common_params_$(arg model).yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find turtlebot3_navigation)/param/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find turtlebot3_navigation)/param/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find turtlebot3_navigation)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(find minitask5)/param/dwa_local_planner_params_$(arg model).yaml" command="load" />
    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    <param name="DWAPlannerROS/min_vel_x" value="0.0" if="$(arg move_forward_only)" />

    <!-- Custom Parameter settings for Planner -->
    <!-- <param name="DWAPlannerROS/max_vel_x" value="0.2"/>
    <param name="DWAPlannerROS/min_vel_x" value="-0.15" if="$(eval arg('move_forward_only') == false)"/>
    <param name="DWAPlannerROS/max_vel_theta" value="2.6"/>
    <param name="DWAPlannerROS/min_vel_theta" value="0.5"/>
    <param name="DWAPlannerROS/acc_lim_x" value="1.8"/>
    <param name="DWAPlannerROS/acc_lim_theta" value="1.8"/> -->
    
  </node>
</launch>
